apis/python/src/tiledbsoma/options/soma_tiledb_context.py

```diff
@@ -39,4 +40,29 @@ class SOMATileDBContext:
     member_uris_are_relative: Optional[bool] = None
     """Allows "relocatability" for local disk / S3, and correct behavior for TileDB Cloud."""
 
-    # read_timestamp, e.g.
+    read_timestamp_start: Optional[int] = None
+    "Timestamp range start for all array read operations. Usually, implicitly zero."
+
+    read_timestamp_end: Optional[int] = None
+    """
+    Timestamp range end for all array write operations.
+    If unspecified, then implictly the latest data as of the first access to each array.
```

Comment:
Q: currently leaving the default timestamp to None meaning no consistency between different arrays by default. If we want consistency by default then we could make the read_timestamp_end default to the time of context initialization. Any opinions?

Reply:
consistency by default supports the principle of least astonishment, so I'm for it.

Reply:
I ran into some rough waters making the read timestamp active by default, as it makes writers unable to read their own writes (thanks @johnkerl for late-night debugging help). Hoping this doesn't blow up my argument to separate the timestamps from open(r/w), but need to noodle on it :sweat_smile: